### PR TITLE
Week 5 joseph han

### PR DESCRIPTION
**Pull request by jhan-dev**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [✅] JavaScript Coding Challenges
- [✅] Ruby Coding Challenges
- [✅] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -10,43 +12,239 @@
...
+    })   
+})
+
+//  After runing yarn jest, I should expect a good fail prompt saying function is not defined
+//      ReferenceError: fullHouseCheck is not defined
 
 // b) Create the function that makes the test pass.
+
+//  To make the test pass, I will need to create the function I am testing for above and pass in an array of 5 nums and to count 2 unique types of cards to have a pair and a 3 of a kind to return true or false if that condition is met.
+const fullHouseCheck = (hand) => {
```
Comment (elysemontano) [inline code-challenges.test.js:201]: Nice work on this problem!  The logic looks great, however, when possible I would always recommend utilizing higher-order functions.

File: interview-questions.md [hunk 2]
```diff
@@ -10,42 +10,88 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 5. What are primary keys? Why are they important?
 
 Your answer:
 
+    - A primary key is used in a relational database. It is a unique id attribute that is needed to have the ability to create, read, update, and delete from the table.
+
+
 Researched answer:
 
+    - A primary key is an attribute from the column of a table in a relational model. The primary key is used as a unique identifier where it is used to find, view specific scope of a data record. Primary keys CANNOT have NULL values, as it either exists in the data table or does not exist.
```
Comment (elysemontano) [inline interview-questions.md:66]: Nice work in this section!  Great work on expanding your answers and providing detail 🎉

File: ruby-challenges.rb [hunk 3]
```diff
@@ -1,32 +1,100 @@
...
+#   PROBLEM #2
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
+# access us_states Hash with .values then use .flatten to remove inner arrays within array then use .sort to alphabetize the values and display with p to print
+p us_states.values.flatten.sort
```
Comment (elysemontano) [inline ruby-challenges.rb:40]: Keep in mind that the prompt is asking for a method.  This logic looks great, a method will make this block of code more dynamic.

**Review by elysemontano — APPROVED** [review]
Great work Joseph! Overall, I am seeing your work improving.  I appreciate that you implemented higher-order functions most of the time in your code.  A suggestion would be to make sure to keep your code dynamic (see comment in Ruby section).  Otherwise, nice work!

File: ruby-challenges.rb (hunk 3 above)
Comment (jhan-dev) [reply ruby-challenges.rb:40]: This portion was completely my fault. I should have seen that key detail but I must have over looked it at the moment from trying to complete the assessment rather quickly. I will make sure to not miss key details like that. I know how to setup Ruby methods (aka functions lmao) I could have easily created that and got that logic in there as well. 

Thank you for the feedback!

**Pull request merged**